File: leetcode/100-999/136. Single Number/solution_1.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var singleNumber = function(nums) 
{
    nums.sort()
    
    for (let i = 0; i < nums.length; i += 2)
    {
        if (i == nums.length - 1)
            return nums[i]

        if (nums[i] != nums[i + 1])
            return nums[i]
    }

    return -1
};

File: leetcode/100-999/136. Single Number/solution_1.cs
public class Solution
{
    public int SingleNumber(int[] nums)
    {
        Array.Sort(nums);

        for (int i = 0; i < nums.Length; i += 2)
        {
            if (i == nums.Length - 1)
                return nums[i];

            if (nums[i] != nums[i + 1])
                return nums[i];
        }

        return -1;
    }
}